plugin-server/src/types.ts

```diff
@@ -833,7 +840,9 @@ export interface JobQueueConsumerControl {
     resume: () => Promise<void> | void
 }
 
-export type IngestEventResponse = { success?: boolean; error?: string; actionMatches?: Action[] }
+export type IngestEventResponse =
```

Comment:
This would probably be cleaner if we always returned everything. And then just set `error` and `preIngestionEvent` to `null`

Reply:
I disagree. Disambiguated unions beat undefineds and others.